disruption_py/inout/mds.py

```diff
@@ -52,13 +50,43 @@ def get_shot_connection(self, shot_id: int):
         return MDSConnection(self.conn, shot_id)
 
 
+def _better_mds_exceptions(func):
+    """
+    Decorator to catch MDSplus exceptions and recreate them with a more descriptive
+    error message which includes the tree and the node path.
+    """
+
+    def wrapper(*args, **kwargs):
+        self = args[0]
+        if len(args) > 1:
+            path = args[1]
+        else:
+            path = kwargs.get("path", None)
+        try:
+            return func(*args, **kwargs)
+        except MDSplus.mdsExceptions.TreeFOPENR:
```

Comment:
I added `TreeFOPENR` and switched order of the subsequent two exceptions to have them ordered hierarchically.